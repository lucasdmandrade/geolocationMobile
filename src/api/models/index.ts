export interface LocationPoint {
  id: string;
  latitude: number;
  longitude: number;
  speed: number;
  time: Date;
}

export interface AllPackagesKeys {
  keys: string[];
}
