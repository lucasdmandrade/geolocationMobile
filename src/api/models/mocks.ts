import { AllPackagesKeys, LocationPoint } from ".";

export const mockPoints: { points: LocationPoint }[] = [
  {
    points: {
      id: "bd6c780f-be9a-453f-9eb3-2d5a9c8f84cb",
      latitude: 37.4219983,
      longitude: -122.084,
      speed: 0,
      time: "2023-11-06T19:02:34.848Z",
    },
  },
  {
    points: {
      id: "968af977-7635-4d11-9b99-e0d405cefc24",
      latitude: 37.4219983,
      longitude: -122.084,
      speed: 0,
      time: "2023-11-06T18:55:35.981Z",
    },
  },
  {
    points: {
      id: "addd45ff-5df5-4534-a147-93932e4721c2",
      latitude: 37.4219983,
      longitude: -122.084,
      speed: 0,
      time: "2023-11-06T19:09:45.016Z",
    },
  },
];

export const pointsKeys: AllPackagesKeys = {
  keys: [
    "bd6c780f-be9a-453f-9eb3-2d5a9c8f84cb",
    "968af977-7635-4d11-9b99-e0d405cefc24",
    "addd45ff-5df5-4534-a147-93932e4721c2",
  ],
};
