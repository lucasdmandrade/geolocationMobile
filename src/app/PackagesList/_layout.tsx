import { Link, Slot } from "expo-router";
import { View, Text, StyleSheet } from "react-native";
import Constants from "expo-constants";
import { colors } from "../../styles/colors";

const Layout = () => (
  <>
    <View style={styles.container}>
      <Text style={styles.goBackTitle}>Voltar</Text>

      <Link style={styles.title} href="/PackagesList">
        Status
      </Link>

      <View style={styles.imaginaryElement} />
    </View>

    <Slot />
  </>
);

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    flexDirection: "row",
    width: "100%",
    backgroundColor: colors.darkBlue,
    padding: 15,
    justifyContent: "center",
  },
  title: {
    flex: 2,
    textAlign: "center",
    fontSize: 18,
    marginBottom: 5,
    color: colors.smoklyWhite,
  },
  goBackTitle: {
    flex: 1,
    fontSize: 18,
    marginBottom: 5,
    color: colors.white,
  },
  imaginaryElement: {
    flex: 1,
  },
});

export default Layout;
