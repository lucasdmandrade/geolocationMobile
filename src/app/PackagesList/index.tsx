import React, { useCallback } from "react";
import { SafeAreaView, StatusBar, StyleSheet } from "react-native";
import PackageList from "./components/PackageList";

const PackagesList = () => {
  return (
    <SafeAreaView style={styles.container}>
      <PackageList />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});

export default PackagesList;
