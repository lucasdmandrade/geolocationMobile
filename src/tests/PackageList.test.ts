jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock")
);

import MockAdapter from "axios-mock-adapter";
import { act, render, waitFor } from "@testing-library/react-native";
import axios from "../services/axios";
import PackagesList from "../app/PackagesList";
import { mockPoints, pointsKeys } from "../api/models/mocks";

const component = PackagesList;

describe("PackagesList render tests", () => {
  let mock: MockAdapter;

  beforeEach(() => {
    mock = new MockAdapter(axios);
  });

  afterEach(() => {
    mock.restore();
  });

  it("PackagesList should be render", async () => {
    act(() => {
      const getall = mock
        .onGet("/points")
        .reply(200, { keys: [mockPoints[0].points.id] });

      const get = mock
        .onGet(`/points/${mockPoints[0].points.id}`)
        .reply(200, mockPoints[0]);
    });

    const root = await waitFor(() => render(component()));

    expect(root).toMatchSnapshot();
  });
});
