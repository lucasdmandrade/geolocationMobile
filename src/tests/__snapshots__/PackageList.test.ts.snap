// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PackagesList render tests PackagesList should be render 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      [
        {
          "id": "bd6c780f-be9a-453f-9eb3-2d5a9c8f84cb",
          "isSynchronized": true,
          "latitude": 37.4219983,
          "longitude": -122.084,
          "speed": 0,
          "time": "2023-11-06T19:02:34.848Z",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderBottomColor": "#8b8b8b",
              "borderBottomWidth": 1,
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 15,
            }
          }
        >
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                {
                  "fontSize": 16,
                  "marginRight": 15,
                }
              }
            >
              Pacote ID: 
              bd6c780f-be9a-453f-9eb3-2d5a9c8f84cb
            </Text>
            <Text
              style={
                {
                  "fontSize": 14,
                }
              }
            >
              Sincronizado
            </Text>
          </View>
          <View>
            <Text
              style={
                {
                  "fontSize": 14,
                }
              }
            >
              16:02
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
